### ISSUE TITLE
RFC: migrate scaffolder plugins to Backstage repo

### ISSUE THREAD

**Issue by antoniobergas** [Parfuemerie-Douglas/scaffolder-backend-module-azure-pipelines#30]
As more and more people is using Backstage and by that all Azure DevOps users are using this scaffolder plugins I want to suggest to migrate them to the Backstage repo.

AFAIK Backstage maintainers already recommend officially this plugins in the documentation, so would make sense to move them so the community can take ownership to keep them update?

What you think?

@MatthiasKainer 
@hevans-dglcom 
@clemensheithecker

**Comment by antoniobergas** [comment]
A few days ago, one of the key contributors to Backstage brought an interesting RFC from the Backstage repository to my attention. You can find it here: https://github.com/backstage/backstage/issues/20266

Currently, there is an ongoing discussion regarding the best solution, but in the interim, it would be highly beneficial to find a way to update both the azure-devops-scaffolder plugins.

If the proposed move outlined in this RFC is contingent on approval within the Backstage RFC, I'd like to suggest two possible approaches:

Grant commit and PR approval permissions to someone who can assist in updating the plugins. I'm more than willing to take on this responsibility if you're comfortable with it.

If, for any reason, you're unable or unwilling to provide these permissions, would it be feasible to consider forking the repository and publishing it as an @backstage package?

I appreciate your consideration and look forward to hearing from you! 😁

**Comment by antoniobergas** [comment]
Hey there!

I've noticed more and more folks are running into issues with this problem. I really want to find a solution that can help everyone out.

I'm up for creating a fork and trying to fix it. If I manage to fix it, I'll publish it under the "@backstage" name.

Down the road, when you've got some time, we can chat about merging these fixes and discuss what steps you'd like to take next.

I hope you understand how important it is to take action. Thanks a bunch!

**Comment by antoniobergas** [comment]
Hi! I created 2 temporary npm packages with the fixes!

[Repositories plugin package](https://www.npmjs.com/package/@antoniobergas/scaffolder-backend-module-azure-repositories)

[Pipeline plugin package](https://www.npmjs.com/package/@antoniobergas/scaffolder-backend-module-azure-pipelines)

I just did a quick fix until we hear from the owners, be aware that could be done better 😅 But looks like its working good!

I tested only on Azure DevOps Server, but as far as I understood the code, it should be working equal for dev.azure.com

I would attend any issues related to this fixes that I did, but until we have news about this situation I will not cover any other issues for now, thanks for your compression

As I can understand that you can be distrust about this I courage you to take a look to my forked repos so you can see the all the changes before use the packages:

[Pipelines plugin forked repo](https://github.com/antoniobergas/scaffolder-backend-module-azure-pipelines)

[Repositories plugin forked  package](https://github.com/antoniobergas/scaffolder-backend-module-azure-repositories)

In addition I would share some links from the official Backstage Discord conversation with one of the main contributors for the azdops stack and the RFC related from the official Backstage repo:

[Azure DevOps issue on Discord](https://discord.com/channels/687207715902193673/1154845617756065932)

[Related RFC in Backstage repo](https://github.com/backstage/backstage/issues/20266#issuecomment-1750637520)

I hope this can help to keep updating Backstage until we find a proper solution.

Cheers!

**Comment by cr0wnf0x** [comment]
@antoniobergas I'm sorry for the very late answer and for leaving the community in the air with their issues. I'll take your suggestion for an internal discussion and will get back to you tomorrow about moving the plugins to the official backstage repo and having it maintained by the community. 

Cheers

**Comment by ZOBHeile** [comment]
@cr0wnf0x @antoniobergas Hi is there any update on the progress?

**Comment by antoniobergas** [comment]
Hi @ZOBHeile I have no update yet, maybe the maintainers can shed some light on the matter 🙏

**Comment by antoniobergas** [comment]
Hi guys! 

Is there any news on this? Will appreciate a response to keep this plugins updated for the community, thank you!

**Comment by Tushti02** [comment]
@cr0wnf0x do we have any update on this as createAzurePipelineAction now fails with latest version upgrade of "@backstage/integration": "^1.6.2"

**Comment by jmezach** [comment]
Note that the Backstage team has started a discussion around a community owned plugins repo. See https://github.com/backstage/community/pull/120 for more details.

**Comment by DanielPerezJensen** [comment]
@antoniobergas  thanks for your fixes that you have made, was going mad trying to fix it myself. 

Would suggest the maintainers to really look at this, current scaffolder does not work without Antonios fixes.

**Comment by Phiph** [comment]
Hey @antoniobergas @cr0wnf0x.

I'm blocked by this and I am considering moving the plugins to the community repo, what should we do about the copyright notice? Should it be the original authors - or are you happy for it to be the same pattern in backstage community repo?

**Comment by antoniobergas** [comment]
Hi @Phiph ,

I cannot anwser you about the copyright as I'm not part of the maintainers of this repos but I agree to move it as I suggested with the original issue.

If I may ask, why are you blocked by this? Did you tested my temporary packages?

**Comment by Phiph** [comment]
Ahhh good point I always get mixed up between the 2 packages, I'll try yours out again :) Thanks! 

Does that mean we're kinda blocked on copyright if we moved the code to backstage community?

**Comment by antoniobergas** [comment]
Not sure about what can be done with this copyrigh but anyways I think the actual code its just http calls with fetch and maybe a rework using the azure devops node api would be great 😁

**Comment by Phiph** [comment]
I've tried integrating your packages into backstage, however it looks like the new backend is providing the Type 'ScmIntegrations' and is not assignable to type 'ScmIntegrationRegistry'.

I'll try and pick this up soon.

**Comment by antoniobergas** [comment]
The packages are not compatible with the new backend system, you need to load them as legacyPlugin

**Comment by awanlin** [comment]
Hi, just wanted to share this Issue I logged in the Backstage Community Plugins repo: https://github.com/backstage/community-plugins/issues/1393. Would very much like to get feedback from the maintainers of this repo before moving forward.

**Comment by awanlin** [comment]
Hi, just following up as I have not heard back?

**Comment by awanlin** [comment]
Hi, just checking in again?

**Comment by awanlin** [comment]
Hi, just a note that we will be moving forward with folding these into an Azure DevOps module with these actions in the Community Plugins repo.
